juju/juju
# Leadership and Lease Pinning - Use Strings Instead of Tags

## Description of change

Entities responsible for pin and unpin requests are passed as strings behind the API boundary. At the API client, they are still passed as tags.

For all pin/unpin usages, the pinned applications are represented as application names instead of application tags.

## QA steps

Ensure leadership pinning, unpinning and expiry work as before:

- Bootstrap to LXD.
- `juju deploy redis`.
- `juju add-unit redis`.
- `juju upgrade-series prepare 0` and confirm.
- `juju upgrade-series prepare 1` and confirm.
- The leader should be redis/0. Run `upgrade-series complete 0`
- Stop the container for machine 0 with `lxc stop {container ID}`.
- Watch `juju status` for a while to ensure that leadership does not change from redis/0, even though the machine is down.
- `juju upgrade-series complete 1`
- After a moment, leadership should change to redis/1.

## Documentation changes

None

## Bug reference

N/A
